program-error/src/lib.rs

```diff
@@ -125,17 +155,18 @@ impl fmt::Display for ProgramError {
 pub trait PrintProgramError {
     fn print<E>(&self)
     where
-        E: 'static + std::error::Error + DecodeError<E> + PrintProgramError + FromPrimitive;
+        E: 'static + PrintProgramError + TryFrom<u32>;
 }
 
+#[cfg(feature = "solana-msg")]
```

Comment:
We can figure this out in follow-up work, but since this isn't actually doing any string formatting, we can probably get by with using the syscall directly and completely avoid pulling in `solana-msg` and `format!`, which is typically the source of bloat.

And that way, we don't need the `solana-msg` feature.

What do you think @febo ?

Reply:
Agree – another option is to use a trait that just converts a `ProgramError` to a `&str`, like [here](https://github.com/anza-xyz/pinocchio/blob/main/sdk/pinocchio/src/program_error.rs#L236-L285).

Reply:
If @febo is fine with it I suggest we just remove the feature and make solana-msg a regular dependency. It causes no measurable increase in build time and I doubt the solana-msg crate is going to have any major version bumps